forte/base_classes/forte_options.cc

```diff
@@ -171,6 +176,59 @@ void ForteOptions::add_double_array(const std::string& label, const std::string&
     add(label, "float_list", py::list(), description);
 }
 
+void ForteOptions::set_options_from_dict(pybind11::dict dict) {
+    for (auto item : dict) {
+        auto label = py::cast<std::string>(item.first);
+        if (exists(label)) {
+            auto obj = py::cast<py::object>(item.second);
+            auto value_type = get(label);
+            // here we convert to the option type and check
+            // for an exception if there is a type mismatch
+            try {
+                if (value_type.second == "bool") {
+                    set(label, py::bool_(obj));
+                }
+                if (value_type.second == "int") {
+                    set(label, py::int_(obj));
+                }
+                if (value_type.second == "float") {
+                    set(label, py::float_(obj));
+                }
+                if (value_type.second == "str") {
+                    std::string s = py::str(obj);
+                    to_upper_string(s);
+                    set(label, py::str(s));
+                }
+                if (value_type.second == "gen_list") {
+                    set(label, obj);
+                }
+                if (value_type.second == "int_list") {
+                    set(label, obj);
+                }
+                if (value_type.second == "float_list") {
+                    set(label, obj);
+                }
+            } catch (std::exception& e) {
+                std::string msg =
+                    "\n  ForteOptions::set_from_dict: the value corresponding to option " + label +
+                    " is not compatible with the option type (" + value_type.second + ")\n" +
+                    "  See the exception:\n" + e.what() + "\n";
+                throw std::runtime_error(msg);
+            }
+        } else {
+            std::string msg =
+                "\n  ForteOptions::set_options_from_dict: option " + label + " is not defined\n";
+            throw std::runtime_error(msg);
+        }
+    }
+}
+
+void ForteOptions::set_dict(const pybind11::dict dict) {
+    dict_ = py::module::import("copy").attr("deepcopy")(dict);
+}
+
+void ForteOptions::reset_dict() { dict_ = pybind11::dict(); }
```

Comment:
Reset could also mean set back to default values.

Reply:
This function does not make any sense (it is equivalent to allocating a new object), so I am going to remove it.